### PR TITLE
Simplify source evaluation options.

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -969,12 +969,12 @@ Perform JavaScript `eval`.
 jerry_value_t
 jerry_eval (const jerry_char_t *source_p,
             size_t source_size,
-            bool is_strict);
+            uint32_t parse_opts);
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md]: If this is an api change please add it to #2213.

File: jerry-core/ecma/operations/ecma-eval.c [hunk 2]
```diff
@@ -82,15 +80,19 @@ ecma_op_eval (ecma_string_t *code_p, /**< code string */
...
 {
 #ifndef JERRY_DISABLE_JS_PARSER
   JERRY_ASSERT (code_p != NULL);
 
   ecma_compiled_code_t *bytecode_data_p;
 
-  bool is_strict_call = (is_direct && is_called_from_strict_mode_code);
+  uint32_t is_strict_call = ECMA_PARSE_STRICT_MODE | ECMA_PARSE_DIRECT_EVAL;
+
+  if ((parse_opts & is_strict_call) != is_strict_call)
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-eval.c]: Are you sure this is correct?

Comment (rerobika) [reply jerry-core/ecma/operations/ecma-eval.c]: I think it's an equivalent conversion. In the previous version the `is_strict_call` variable was only true, if both options were true. So it means whether one of the `ECMA_PARSE_STRICT_MODE` or ` ECMA_PARSE_DIRECT_EVAL` flags are not enabled, the script must be executed in non-strict mode.

File: docs/02.API-REFERENCE.md (hunk 1 above)
Comment (rerobika) [reply docs/02.API-REFERENCE.md]: Thanks I've added it.

File: jerry-core/ecma/operations/ecma-eval.c (hunk 2 above)
Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-eval.c]: Ok

**Pull request merged**